homeassistant/components/freedompro/fan.py

```diff
@@ -87,10 +88,15 @@ async def async_added_to_hass(self) -> None:
         await super().async_added_to_hass()
         self._handle_coordinator_update()
 
-    async def async_turn_on(self, percentage=None, preset_mode=None, **kwargs):
+    async def async_turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
         """Async function to turn on the fan."""
-        payload = {"on": True}
-        payload = json.dumps(payload)
+        payload_dict = {"on": True}
+        payload = json.dumps(payload_dict)
```

Comment:
`mypy` does not like reassigning from a `dict` to a `str`, so I've created an intermediate variable.